src/ui/components/ShowMoreCard/index.js

```diff
@@ -1,55 +1,82 @@
+/* @flow */
 /* eslint-disable react/no-danger */
-
+import invariant from 'invariant';
 import makeClassName from 'classnames';
 import * as React from 'react';
-import PropTypes from 'prop-types';
 import { compose } from 'redux';
 
 import translate from 'core/i18n/translate';
+import withUIState from 'core/withUIState';
 import { sanitizeHTML } from 'core/utils';
 import Card from 'ui/components/Card';
+import type { I18nType } from 'core/types/i18n';
 
 import './styles.scss';
 
 // This refers to height of `Card-contents` CSS class,
 // beyond which it will add read more link.
 export const MAX_HEIGHT = 150;
 
-export class ShowMoreCardBase extends React.Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    header: PropTypes.node,
-    i18n: PropTypes.object.isRequired,
-  };
+type UIStateType = {|
+  expanded: boolean,
+|};
 
-  constructor(props) {
-    super(props);
-    this.state = { expanded: true };
-  }
+type Props = {|
+  children: React.Element<any> | string,
+  className?: string,
+  header?: React.Element<any> | string,
+  id: string,
+|};
 
-  componentWillReceiveProps() {
-    this.truncateToMaxHeight(this.contents);
```

Comment:
To follow up on our conversation, I'm wondering whether you can address the issue by just adding a check into the existing `componentWillReceiveProps` that checks whether `children` has changed, and only calls `this.truncateToMaxHeight` is `children` has in fact changed.

Reply:
As mentioned I did something similar but it wasn't working. I revisited this just to be sure but I am running into the same issue where the height is not accurate when it first see there is a difference (if that makes sense). This was the same issue that I was running into with adding this into just componentDidMount. Here's the full code that I am testing now. Maybe you will see something I am not tho:

https://pastebin.com/GeUt3BHp